### ISSUE TITLE
GenevaERS TSC Meeting - Aug 26, 2020

### ISSUE THREAD

**Issue by AOrthVector** [genevaers/community#13]
Meeting notes from the TSC held on Aug 125, 2020. All interested in GenevaERS development are welcome to join.

# Agenda:

Proposed GenevaERS 5.0 Architectural Direction  
Logo Adoption and Marketing Update  
Shared System Access  
Getting Started Guides  
Joining the Project  
GenevaERS Installation and Use  
Source Code Release, Evaluation Version and zIIP removal  
Spark POC  
Round table  

## Conference call details

Join Zoom Meeting
https://zoom.us/j/94966206199

Meeting ID: 949 6620 6199

## Attendance
Voting member rollcall:  
[x ] Randall Ness, IBM (RN)  
[x ] Jeff Horner, IBM  
[x ] Gillian Hannington, IBM  
[x ] Ian Cunningham, IBM (IC)  
[ ] Eugene Morrow, IBM  
[x ] Bob McCormack, IBM  
[x ] Kip Twitchell, IBM (KT)  

## Other Attendees
Jim Hladyshewsky, SAFR Delivery Manager, IBM  
Andrea Orth (AO)  
Sandy Peresie  

# Minutes

## Proposed GenevaERS 5.0 Architectural Direction

Randall presented his slidedeck which captured:  
* Current State
* Pros and cons of remaining on current GenevaERS architecture
* Possible Future State; GenevaERS 5

A great deal of discussion occured around future state. 

A proposed first iteration of GenevaERS 5 would be to keep the backend "as is"; C++, COBOL, and Assembler while having a more modern front end coded using a "variant" of the open source language Groovy; aka "Groovy Geneva".  

In this solution, the free software code editor VSCode would become the IDE for the new "GG Studio". GG Studio is a separate entity from the proposed "GG Source Code".  Using current state terminology, GG Source Code would be such things as the LRs, Lookup Paths, etc. The run control files; JLT, XLT, VDP creation would be initiated by the front end. Open source tooling Gradle would be the technology behind the scenes creating the run control files.

Future iterations would include the capability of custom front ends. A simple synopsis was "All the power of Geneva with whatever front end you want".  

There was a callout to Open Mainframe Project [Polycephaly](https://www.openmainframeproject.org/projects/polycephaly). Polycephaly is built using Groovy and provides developers modern tooling options. It removes the need for separate development paths for distributed and z/OS workloads. Developers can develop on any platform, store to Git and use Jenkins to deploy.

This project has had some preliminary discussions with the Polycephaly project.

The action item for this Geneva 5 discussion is to setup a cadence for a series of architectural meetings.  

## Logo Adoption and Marketing Update  
Multiple people liked / could go with concept 1A; options can be seen in the GenevaERS Slack channel. Sandy would like the logo slanted forward to indicate speed. Action item after the meeting was to ask John Metric if we could have the logo modified to slant.  

## Shared System Access  

Vicom Infinity is a sponsor of the Open Mainframe Project and has offered a shared LPAR for us to use. The LPAR has 2 processors and may not have DB2 started. Members of this project may need to help getting DB2 up and running on this LPAR.

5 user IDs were requested today for this instance. It may be next week before we get access. We will also be given a VPN to get into Vicom Infinity's network. We could then connect the current version of the SAFR/Geneva workbench via JDBC and run the batch workflow.  

## Joining the Project  
* Sandy is working on a Joining the Project Guide to aid in demystifying all the signups, etc that is necessary to follow / contribute for the project. 
* Andrea added a new menu option called Community on the GenevaERS website. The menu contains options for the mailing list, Calendar, and GitHub Community repository.

Kip did a short explainer of the two sets of repositories; IBM and GenevaERS. The repositories under IBM are staging repositories. One purpose is to perform scans to find potential licensing issued for code is transferred to the GenevaERS repository.

## Request for the next TSC meeting

Kip would like to have an uplifting, hopeful opener for each TSC meeting

**Comment by bobmcc9** [comment]
added Meeting label

**Comment by KipTwitchell** [comment]
We need to get these saved into the meeting directory, so I have reopened the ticket.

**Comment by KipTwitchell** [comment]
@AOrthVector, these are good notes; but I had the same problem last time when Bob put them in an issue.  I can't find a simple tool that will take this HTML and save it as Markdown, which is what I think our notes should be stored in.  I had to hand convert them in Atom.  I just tried to use Word which will save as HTML, but I couldn't find a simple tool to make them Markdown.  

I think perhaps actually in the future simply doing a branch of the community repo, adding a new text file as mark down under the TSC/Meeting directory,  Adding the file to the branch, committing (along with signing) and pushing the branch, creating the PR might be how we should do this.  

Happy to discuss if there is another way.

**Comment by AOrthVector** [comment]
That's odd that you had to convert it. It's markdown. But I had to confirm GitHub didn't do something funny, so I put the issue in edit mode to see the markdown.

It looks like the repo came with some pre-defined GitHub actions, we get emailed TSC reminders from one action. Wonder if the one for meeting minutes doesn't work correctly, or if the instructions for it aren't correct. 

Next time, I'll just do the 'ol pull request thing. I set up a SSH key last night, and I might be ready to fork, but I need to look at the documentation some more.  Some documentation may come out of all this fumbling around
